tile_pages.py

```diff
@@ -255,79 +275,135 @@ def run(self,rows=None,cols=None):
         # We need to account for the shift in origin if page rotation is applied
         o_shift = [0,0]
 
-        if self.rotation != 0:
+        if self.rotation != SW_ROTATION_NONE:
             # define the rotation transform and
             # swap the trim order
-            if self.rotation == 1:
+            if self.rotation == SW_ROTATION_CLOCKWISE:
                 R = [0,-1,1,0]
-                o_shift = [0,pw]
+                o_shift = [0,pw[0]]
                 order = [3,2,0,1]
 
-            if self.rotation == 2:
+            if self.rotation == SW_ROTATION_COUNTERCLOCKWISE:
                 R = [0,1,-1,0]
-                o_shift = [ph,0]
+                o_shift = [ph[0],0]
                 order = [2,3,1,0]
-                       
+            
             # swap width and height of pages
-            tmp = ph
-            ph = pw
-            pw = tmp
+            ph, pw = pw, ph
             
             trim = [trim[o] for o in order]
         
-        # define the output page media box
-        width = (pw - trim[0] - trim[1])*cols
-        height = (ph - trim[2] - trim[3])*rows
+        row_height = [0]*self.rows
+        width = 0
+        height = 0
+        
+        for r in range(self.rows):
+            width = max(width,sum(pw[r*self.cols:r*self.cols+self.cols]))
+            row_height[r] = max(ph[r*self.cols:r*self.cols+self.cols])
+        width -= (trim[0] + trim[1])*self.cols
+        height = sum(row_height) - (trim[2] + trim[3])*self.rows
+                
+        if self.target_width and self.target_height:
+            width = self.target_width
+            height = self.target_height
+            row_height = [self.target_width for i in range(self.rows)]
+            page_box_width = self.target_width / self.cols
+            page_box_height = self.target_height / self.rows
+            page_box_defined = True
+        else:
+            page_box_defined = False
+                
         media_box = [0,0,width + 2*margin,height + 2*margin]
-
-        if media_box[2] > 14400 or media_box[3] > 14400:
-            print ('**************************************')
-            if self.units == 1:
-                print(_('Warning! Output is larger than 508 cm, may not open correctly.'))
-            else:
-                print(_('Warning! Output is larger than 200 in, may not open correctly.'))
-            print ('**************************************')
+        
+        max_size_px = 14400
+        if media_box[2] > max_size_px or media_box[3] > max_size_px:
+            print (62 * '*')
+            print(_(f'Warning! Output is larger than {round(self.px_to_units(max_size_px))} {unitstr}, may not open correctly.'))
```

Comment:
f strings don't play nicely with gettext, so this needs to be either the old style .format strings or hardcoded as before.

Reply:
Okay, I think I'll change it to .format()